core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -472,6 +472,47 @@ class DAGScheduler(
   }
 
   /**
+   * Get ancestor splits in ShuffledRDD.
+   */
+  private[spark] def parentSplitsInShuffledRDD(stageId: Int, pId: Int): Option[Map[Int, Set[Int]]] =
+  {
+    stageIdToStage.get(stageId) match {
+      case Some(stage) =>
+        val waitingForVisit = new Stack[Tuple2[RDD[_], Int]]
+        waitingForVisit.push((stage.rdd, pId))
+        val ret = new HashMap[Int, HashSet[Int]]()
+        while(waitingForVisit.nonEmpty) {
+          val (rdd, split) = waitingForVisit.pop()
+          if (getCacheLocs(rdd)(split) == Nil) {
+            rdd.dependencies.foreach {
+              case dep: ShuffleDependency[_, _, _] =>
+                val noPartitionerConflict = rdd.partitioner match {
+                  case Some(partitioner) =>
+                    partitioner.isInstanceOf[HashPartitioner] &&
+                    dep.partitioner.isInstanceOf[HashPartitioner] &&
+                    partitioner.numPartitions == dep.partitioner.numPartitions
```

Comment:
Yes, I always think rdd.partitioner should be the same with shuffle dependencies partitioner. But I found `CustomShuffledRDD` is a different one.